lib/kafka/async_producer.rb

```diff
@@ -148,14 +151,12 @@ def ensure_threads_running!
       @timer_thread ||= Thread.new { @timer.run }
     end
 
-    def buffer_overflow(topic)
+    def buffer_overflow(topic, message)
       @instrumenter.instrument("buffer_overflow.async_producer", {
         topic: topic,
       })
 
-      @logger.error "Cannot produce message to #{topic}, max queue size (#{@max_queue_size}) reached"
```

Comment:
I'd like to keep the log statement.

Reply:
I removed it because the sync producer doesn't have it either, and the exception already has the message. If you want exception AND logging, and I suggest we add the logging to the sync producer as well for consistency. Should I do that?